Added Swamp endvault with Fedhas and Yred theming.

ByrelMitchell opened the pull request.

gammafunk commented:
Thanks for the submission. This is a reasonable idea for an end vault, although it does share theme with the other undead map. I think it's interesting to try some yred-themed monster "crossover" from Crypt like you've done, although getting a good set of monsters for that might take some work. I can leave some quick comments now and take a closer look later.

ByrelMitchell commented on crawl-ref/source/dat/des/branches/swamp.des:
```diff
@@ -2361,3 +2361,63 @@ ttccccC'''cC'-'Cc'''1'1cctt
+ccc........cccE2E3333EE2Eccc               
+ccP....o...PccEE53335333Eccc       
+cct6..P_P.6tcccE3333q335Eccc       
+cc6.P.7P7.P6cccE3333D333Eccc     
```
I scattered a few throughout, including some deliberate dead and living trees spreading toward the other lobe. It should be tricky at this point to get a tile away from all trees without going deep into the lobes unless you luck out and get a wall-heavy front section.

ByrelMitchell commented:
Squashed the commits into one, and reworked the monster list (especially on the Yred side.) Now has a pair of vampire knights with an assortment of yred gifts and undead hydras for backup, and a 1:11 chance of having an undead lerny.

amalloy commented on crawl-ref/source/dat/des/branches/swamp.des:
```diff
@@ -2361,3 +2361,67 @@ ttccccC'''cC'-'Cc'''1'1cctt
+ORIENT:  float
+TAGS:    no_rotate
+MONS:    swamp dragon / swamp drake / hydra
+MONS:	 freezing wraith / flayed ghost / phantasmal warrior / ghoul
```
Whitespace is off here.

ByrelMitchell replied:
> Cool idea. I'm not much of an expert on vault design / balance, so I'll leave that for gammafunk, but I have a couple internal-cleanup things to suggest.

All right, I made those changes. I doubt gammafunk is interested in reviewing this any further; is there anyone else on the dev team that typically reviews vault game design stuff?

amalloy commented on crawl-ref/source/dat/des/branches/swamp.des:
```diff
@@ -2361,3 +2361,67 @@ ttccccC'''cC'-'Cc'''1'1cctt
+KMONS:   p = plant
+KMONS:   ' = oklob plant
+KMONS:   q = vampire knight
+: if crawl.one_chance_in(11) then
```
A few questions here. This vault will occur in like 7% of Swamp:$ instances, given how many other Swamp:$ vaults there are. Placing the undead Learnaean hydra in just 9% of those will make it a pretty rare find. I imagine that's intentional, I'm just a little surprised you wanted it to be so rare. Do you think it's too difficult to place 100% of the time? Do you want it to be so rare as to be very exciting when a player runs into it?

Second, I wouldn't do this with a one_chance_in. Why not just do it with a weighted MONS, where the sum of weights for undead lernaean hydrae is much less than the total weight?

    MONS: the lernaean hydra skeleton / spectral the lernaean hydra / ... / spectral hydra w:100 / hydra simulacrum w:100

ByrelMitchell replied:
For both of these, I was copying swamp_basic_johnstein; now that I look more carefully, I see that one is an outlier. I'm pretty sure undead lerny is significantly easier than non-undead-lerny; low MR, and the only high damage one is also slow.

I'll change this.

amalloy commented on crawl-ref/source/dat/des/branches/swamp.des:
```diff
@@ -2361,3 +2361,67 @@ ttccccC'''cC'-'Cc'''1'1cctt
+MONS:    patrolling thorn hunter / patrolling shambling mangrove
+MONS:    patrolling spriggan druid god:fedhas
+KITEM:   O = decaying rune of Zot
+KFEAT:   _ = altar_fedhas
```
`_` and `D` are symbols that don't have any obvious similarity or mnemonic. What about `F` and `Y` to label each lobe, or `_` and `-` to make them both look altar-ish?

ByrelMitchell replied:
Fair enough

ByrelMitchell commented:
> Cool idea. I'm not much of an expert on vault design / balance, so I'll leave that for gammafunk, but I have a couple internal-cleanup things to suggest.

All right, I made those changes. I doubt gammafunk is interested in reviewing this any further; is there anyone else on the dev team that typically reviews vault game design stuff?

amalloy commented:
Merged as 8ac039397e, after some small fixes to commit message and comments. Thanks for the vault!
